frame/multisig/src/migrations.rs

```diff
@@ -0,0 +1,69 @@
+use super::*;
+use frame_support::{
+	dispatch::GetStorageVersion,
+	traits::{OnRuntimeUpgrade, WrapperKeepOpaque},
+	Identity,
+};
+
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+
+pub mod v1 {
+	use super::*;
+
+	type OpaqueCall<T> = WrapperKeepOpaque<<T as Config>::RuntimeCall>;
+
+	#[frame_support::storage_alias]
+	type Calls<T: Config> = StorageMap<
+		Pallet<T>,
+		Identity,
+		[u8; 32],
+		(OpaqueCall<T>, <T as frame_system::Config>::AccountId, BalanceOf<T>),
+	>;
+
+	pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<(), &'static str> {
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			ensure!(onchain < 1, "this migration can be deleted");
+
+			log!(info, "Number of calls to refund and delete: {}", Calls::<T>::iter().count());
+
+			Ok(())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			if onchain > 0 {
+				log!(info, "MigrateToV1 should be removed");
+				return T::DbWeight::get().reads(1)
+			}
+
+			// TODO: Assuming this is one write and a read per record
+			Calls::<T>::drain().for_each(|(_call_hash, (_data, caller, deposit))| {
+				//TODO: What's the weight of one unreserve?
+				T::Currency::unreserve(&caller, deposit);
+			});
+
+			current.put::<Pallet<T>>();
+
+			<T as frame_system::Config>::BlockWeights::get().max_block
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade() -> Result<(), &'static str> {
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
```

Comment:
Could you add an assertion that the new version is exactly `1`?

Reply:
Yeah, the assertion we have now is wrong, about to fix it.